CHANGELOG.md

```diff
@@ -1,10 +1,43 @@
-## vNext (TBD)
+## 12.0.0-alpha.0 (2023-03-21)
 
-### Deprecations
-* None
+This is the first pre-release of the next major version of our SDK.
+Please read more and discuss in the dedicated discussion: https://github.com/realm/realm-js/discussions/5416
+
+### Breaking changes
+
+Although this is a complete re-write of our SDK, we've strived to keep breakages to a minimum and expect our users to upgrade from v11 without any notible changes to their code-base.
+
+* In an effort to align with EcmaScript modules, we’re adopting a pattern of named exports on the “root” of the package namespace. This change also affects how our library is imported on CommonJS / Node.js: Before, we exported our constructor directly from the package (`module.export = Realm` style), which would allow CommonJS runtimes like Node.js to import us using a simple assignment:
+```javascript
+const Realm = require(“realm”); // this commonjs style - won’t work anymore
+From now on, users need to consume us using a named import, like this:
+const { Realm } = require(“realm”); // commonjs style
+import { Realm } from “realm”; // esm style
+import Realm from “realm”; // esm style - consuming our “default” export
+```
+* Similarly we’re deprecating our namespaced API (the long chaining of identifiers, such as Realm.App.Sync.setLogLevel) in favor of the shorter named exports. Our main motivation is that it has proven very verbose and hard to maintain in our new SDK and since its inception (which predates ES Modules), we believe the community has moved towards a preference for simple named exports. We would love your feedback on this decision - please comment on the discussion linked above 👆
+* The entire BSON package used to be re-exported as Realm.BSON, to simplify the new SDK we want to export only the BSON types that our SDK database component supports (ObjectId, Decimal128 and UUID). See #4934.
+* We're now reusing code to perform assertions and although this is strictly not a breaking change, since we havn't historically documented error messages, you should probably revisit any code in your app which relies on matching on specific error messages.
```

Comment:
```suggestion
* We're now reusing code to perform assertions and although this is strictly not a breaking change, since we haven't historically documented error messages, you should probably revisit any code in your app which relies on matching on specific error messages.
```